### PR TITLE
Update/solidification/weak solidification

### PR DESCRIPTION
**Pull request by apenzk**
add special rules for weak solidification

### CODE REVIEW

**Review by oliviasaa — APPROVED** [review]

**Comment by oliviasaa** [comment]
I still think that the definitions on this file are misleading; they seem to me that a message is only solid after the booking checks, since we say that 
solid => valid and that 
valid => respects some conditions checked in the booker

**Comment by jkrvivian** [comment]
> I still think that the definitions on this file are misleading; they seem to me that a message is only solid after the booking checks, since we say that
> solid => valid and that
> valid => respects some conditions checked in the booker

A message is considered solid based on **its parents'** `solid` and `valid` status.
But you're right, the description here is somehow confusing that I should improve. :cry:

**Comment by jkrvivian** [comment]
@apenzk since you introduced a new parameter `isStrongRequest`, we should also add it to the `SolidificationRequest` below.

**Comment by apenzk** [comment]
> 
> 
> I still think that the definitions on this file are misleading; they seem to me that a message is only solid after the booking checks, since we say that
> solid => valid and that
> valid => respects some conditions checked in the booker

@oliviasaa I tried to rewrite it, but i am not sure i understand the problem. If you have a good idea how to fix it please go ahead.

**Comment by apenzk** [comment]
> 
> 
> > I still think that the definitions on this file are misleading; they seem to me that a message is only solid after the booking checks, since we say that
> > solid => valid and that
> > valid => respects some conditions checked in the booker
> 
> A message is considered solid based on **its parents'** `solid` and `valid` status.
> But you're right, the description here is somehow confusing that I should improve. 😢

@jkrvivian same to you. i am not sure if my rewrite improved the situation.

**Comment by apenzk** [comment]
> 
> 
> @apenzk since you introduced a new parameter `isStrongRequest`, we should also add it to the `SolidificationRequest` below.

isStrongRequest is variable only kept locally it is not part of any message.

File: 4.4 Solidification.md [hunk 1]
```diff
@@ -12,21 +12,40 @@ In that way, the Tangle enables all nodes to retrieve all of a message's history
 
 ## 4.4.2 Definitions
 * **valid**: A message is considered valid if it passes the following filters from the solidifier and from the message booker:
-    * solidifier: it checks if parents are valid,
-    * booker: it check if the contained transaction is valid. Notice that only messages containing a transaction are required to perform this check, which are defined in [Section 5.1 - UTXO](./5.1%20UTXO.md#validation)
+    * Solidifier: checks if parents are valid,
+    * Booker: checks if the contained transaction is valid. Note that only messages containing a transaction are required to perform this check, which are defined in [Section 5.1 - UTXO](./5.1%20UTXO.md#validation)
```
Comment (oliviasaa) [inline 4.4 Solidification.md:16]:
```suggestion
    * Solidifier: checks if parents are valid,
    * Booker: checks if the contained transaction is valid. Note that only messages containing a transaction are required to perform this check, which are defined in [Section 5.1 - UTXO](./5.1%20UTXO.md#validation)
A message validity status is initialized as *valid*, and it's set as *invalid* whenever the message does not pass any of the filters above.    
```

File: 4.4 Solidification.md [hunk 2]
```diff
@@ -12,21 +12,40 @@ In that way, the Tangle enables all nodes to retrieve all of a message's history
...
 * **parents age check**: A check that ensures the timestamps of parents and child are valid, following the details defined in [Section 4.2 - Timestamps](./4.2%20Timestamps.md).
-* **solid**: A message is solid if it passes parents age check and all its parents are stored in the storage, solid and valid.
+* **solid**: A message is solid if it passes the solidification check and all its parents are stored in the storage, and are solid and valid.
```
Comment (oliviasaa) [inline 4.4 Solidification.md:18]:
```suggestion
* **solid**: A message is solid if it passes the parent age check and all its parents are stored in the storage, solid and their validity status is set as valid.
```

**Comment by oliviasaa** [comment]
I left a suggestion in the definitions section. Let me know if you think it is correct :)

**Pull request merged**